notifications/models.py

```diff
@@ -0,0 +1,52 @@
+from django.db import models
+
+# Create your models here.
+from typing import List, TypedDict
+import uuid
+
+
+class SubscriberJson(TypedDict):
+    provider_id: str
+    subscriber_id: str
+
+
+class EventJson(TypedDict):
+    event_id: str
+    event_provider: str
+    created_at: str
+
+
+class NotificationJson(TypedDict):
+    event: EventJson
+    subsriber: SubscriberJson
+    seen: bool
+
+
+class Event(models.Model):
+    event_id = models.CharField(max_length=200)
+    event_provider = models.CharField(max_length=200)
+    created_at = models.DateTimeField()
+
+    def to_json(self) -> EventJson:
+        return {
+            "event_id": self.event_id,
+            "event_provider": self.event_provider,
+            "created_at": self.created_at.isoformat(timespec="seconds"),
+        }
```

Comment:
I don't think you are going to need that in this form.

I'd remove all `to_json` for now and add them when you need them (i.e. when creating the API). Then you also know in which form you need them